src/ResourceManager/Consumption/Commands.Consumption.Test/ScenarioTests/BudgetTests.ps1

```diff
@@ -0,0 +1,237 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+New budget
+#>
+function Test-NewBudget
+{
+    $budget = New-AzureRmConsumptionBudget -Amount 60 -Name PSBudget -Category Cost -StartDate 2018-06-01 -EndDate 2018-11-01 -TimeGrain Monthly
+	Assert-NotNull $budget
+	Assert-AreEqual 60 $budget.Amount
+	Assert-AreEqual PSBudget $budget.Name
+	Assert-AreEqual Cost $budget.Category
+	Assert-AreEqual Monthly $budget.TimeGrain
+}
+
+<#
+.SYNOPSIS
+New budget at resource group level
+#>
+function Test-NewBudgetAtResourceGroupLevel
+{
+    $budget = New-AzureRmConsumptionBudget -ResourceGroupName RGBudgets -Amount 60 -Name PSBudgetRG -Category Cost -StartDate 2018-06-01 -EndDate 2018-11-01 -TimeGrain Monthly 
```

Comment:
We would like to hard code resource group name `RGBudgets` instead of depending on another cmdlet Get-AzureRmResourceGroup to verify budget test. Even created under test subscription, API works if resource group name is not under the subscription. Rerecording can proceed, although current spend in this situation will be null.